assets/js/main.js

```diff
@@ -71,30 +72,206 @@ $(document).ready(function() {
+          $(this).attr('target', '_blank')
+      }
+  })
+
+  authenticateUser()
+
+  async function authenticateUser() {
+    const { response, json } = await window.userRequest
+    window.auth = { loaded: true }
+
+    if (response && response.ok && json && json.user) {
+        Object.assign(window.auth, json.user.user)
+        if (json.user.ok) {
+            window.loggedIn = true
+            window.auth._id = json.user._id
+            $('[data-js-header-auth-user]').addClass('transition-in')
+            $('[data-js-header-user-avatar]').html('<img src="' + auth.avatarUrl + '" alt=""/>')
+            activeUserChanges()
+        } else {
+            window.loggedIn = false
+            $('[data-js-header-auth-visitor]').addClass('transition-in')
+            visitorChanges()
+        }
+    } else {
+        window.loggedIn = false
+        $('[data-js-header-auth-visitor]').addClass('transition-in')
+        visitorChanges()
+    }
+    $('[data-js-header-auth-loading]').addClass('transition-out')
+    setTimeout(function() {
+        $('[data-js-header-auth-loading-grey-line]').hide()
+    }, 1000)
+  }
+
+  $('[data-js-header-auth-logout]').on('click', userLogout)
+
+  async function userLogout() {
+    $('[data-js-header-authentication]').addClass('loading')
+    $('[data-js-header-auth-loading]')
+        .removeClass('transition-out')
+        .addClass('transition-in')
+    $('[data-js-header-auth-loading-grey-line]').show()
+    if ($(window).innerWidth() < 841) {
+        $('[data-js-header-menu-toggle]').removeClass('open')
+        $('[data-js-header-menu-wrap]').slideUp(200)
+    }
+    $('[data-js-header-menu-wrap]').slideUp(200)
+
+    const url = '{{ site.Param "backendURL" }}/logout'
+    const options = {
+        method: 'GET',
+        mode: 'cors',
+        credentials: 'include',
+        headers: { Accept: 'application/json' },
+    }
+    try {
+        await fetch(url, options)
+        $('[data-js-header-auth-user]')
+            .removeClass('transition-in')
+            .addClass('transition-out')
+        setTimeout(function() {
+            $('[data-js-header-auth-loading-grey-line]').hide()
+            $('[data-js-header-auth-visitor]')
+                .removeClass('transition-out')
+                .addClass('transition-in')
+        }, 1000)
+        visitorChanges()
+    } catch (error) {
+        location.reload()
+    } finally {
+        auth = { loaded: true }
+        window.loggedIn = false
+        $('[data-js-header-authentication]').removeClass('loading')
+        $('[data-js-header-auth-loading]')
+            .removeClass('transition-in')
+            .addClass('transition-out')
+    }
+  }
+
+  function activeUserChanges() {
+    $(document).ready(function() {
+        $('[ad-type*="product"]').hide()
+    })
+    $('#footer-cta .btn')
+        .text('Return to app')
+        .attr('href', '{{ site.Param "appURL" }}/apps')
+  }
+  function visitorChanges() {
```

Comment:
Elements for this function do not exist in documentation. Remove function and it's call.